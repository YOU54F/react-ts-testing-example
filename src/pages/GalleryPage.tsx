import * as React from 'react';
import { GridDisplay } from '../components/GridDisplay/GridDisplay';

export class Gallery extends React.Component<{}, {}> {
  public render() {
    return (
      <div className='gallery-page-container' id='gallery_grid'>
        <GridDisplay
          gridItems={[
            'https://picsum.photos/100/150',
            'An',
            'https://picsum.photos/100/150',
            ' ',
            'https://picsum.photos/100/150',
            ' ',
            'https://picsum.photos/100/150',
            'Example',
            'https://picsum.photos/100/150',
            ' ',
            'https://picsum.photos/100/150',
            ' ',
            'https://picsum.photos/100/150',
            'Website',
            'https://picsum.photos/100/150'
          ]}
        />
      </div>
    );
  }
}
