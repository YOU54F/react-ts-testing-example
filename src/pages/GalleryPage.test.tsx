import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { Gallery } from './GalleryPage';

describe('Gallery Page', () => {
  it('renders the gallery grid', () => {
    expect(shallow(<Gallery />).find('#gallery_grid').length).toEqual(1);
  });

  it('can render the Gallery page', () => {
    const tree = renderer.create(<Gallery />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
