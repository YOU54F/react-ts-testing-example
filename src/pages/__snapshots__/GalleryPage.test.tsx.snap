// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Page can render the Gallery page 1`] = `
<div
  className="gallery-page-container"
  id="gallery_grid"
>
  <div
    className="grid-display-container"
  >
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "0s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-0"
        src="https://loremflickr.com/100/150"
      />
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "0.5s",
        }
      }
    >
      <p
        className="grid-text"
      >
        An
      </p>
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "1s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-2"
        src="https://loremflickr.com/100/150"
      />
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "1.5s",
        }
      }
    >
      <p
        className="grid-text"
      >
         
      </p>
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "2s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-4"
        src="https://loremflickr.com/100/150"
      />
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "2.5s",
        }
      }
    >
      <p
        className="grid-text"
      >
         
      </p>
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "3s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-6"
        src="https://loremflickr.com/100/150"
      />
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "3.5s",
        }
      }
    >
      <p
        className="grid-text"
      >
        Example
      </p>
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "4s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-8"
        src="https://loremflickr.com/100/150"
      />
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "4.5s",
        }
      }
    >
      <p
        className="grid-text"
      >
         
      </p>
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "5s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-10"
        src="https://loremflickr.com/100/150"
      />
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "5.5s",
        }
      }
    >
      <p
        className="grid-text"
      >
         
      </p>
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "6s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-12"
        src="https://loremflickr.com/100/150"
      />
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "6.5s",
        }
      }
    >
      <p
        className="grid-text"
      >
        Website
      </p>
    </div>
    <div
      className="grid-item fade-out fade-in"
      style={
        Object {
          "animationDelay": "7s",
        }
      }
    >
      <img
        alt="grid-image"
        className="grid-image"
        id="grid-item-14"
        src="https://loremflickr.com/100/150"
      />
    </div>
  </div>
</div>
`;
